### PR TITLE
Resolve environment variables at compile time

### PR DESCRIPTION
**Pull request by benny-n**
This PR's main purpose is to utilize the `env!` macro to resolve environment variables at compile time. 
This means that if a required env var is missing, the build will _fail to compile_, as opposed to what we had so far, which was only a run-time check, resulting in a _run-time error_ instead.
As a bonus, by using a `build.rs` script (which allows us to modify how `cargo` builds the project) we can also get rid of all the annoying `dotenv().ok()` everywhere in the codebase

- Also removed some unused dependencies

### CODE REVIEW

File: packages/server/src/config.rs [hunk 1]
```diff
@@ -1,31 +1,28 @@
...
 #[derive(Debug, Clone)]
-pub struct Config<'cfg> {
-    pub ip: &'cfg str,
-    pub port: u16,
-    pub uri: &'cfg str,
-    pub client_id: &'cfg str,
-    pub profile: &'cfg str,
+pub struct Config {
+    pub ip: &'static str,
+    pub port: &'static str,
```
Comment (liadaram1) [inline packages/server/src/config.rs:10]: Can't we get the value from env! as u16 instead of parsing it later?

File: packages/server/src/error.rs [hunk 2]
```diff
@@ -31,6 +30,22 @@ impl From<bson::oid::Error> for AppError {
     }
 }
 
+impl std::fmt::Display for AppError {
```
Comment (liadaram1) [inline packages/server/src/error.rs:33]: Why do you implement Display trait instead of using the existing one?
It doesn't seem related to the change of the env! macro.

Comment (benny-n) [reply packages/server/src/error.rs:33]: You're right, it isn't related, I got rid of the `#derive` as a part of the "Also removed some unused dependencies" changes, we brought a whole crate for a deriving single trait for a single struct, which could just be simply implemented in less than 10 lines. This is one of those cases where I think bringing a crate isn't necessary, for similar reasons we discussed some time ago about the builder crates.

**Review by liadaram1 — APPROVED** [review]
